packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -1031,58 +982,223 @@ class RawGestureDetectorState extends State<RawGestureDetector> {
       final List<String> gestures = _recognizers.values.map<String>((GestureRecognizer recognizer) => recognizer.debugDescription).toList();
       properties.add(IterableProperty<String>('gestures', gestures, ifEmpty: '<none>'));
       properties.add(IterableProperty<GestureRecognizer>('recognizers', _recognizers.values, level: DiagnosticLevel.fine));
+      properties.add(DiagnosticsProperty<bool>('excludeFromSemantics', widget.excludeFromSemantics, defaultValue: false));
+      properties.add(DiagnosticsProperty<SemanticsGestureDelegate>('semantics', widget.semantics));
     }
     properties.add(EnumProperty<HitTestBehavior>('behavior', widget.behavior, defaultValue: null));
   }
 }
 
+typedef _AssignSemantics = void Function(RenderSemanticsGestureHandler);
+
 class _GestureSemantics extends SingleChildRenderObjectWidget {
   const _GestureSemantics({
     Key key,
     Widget child,
-    this.owner,
-  }) : super(key: key, child: child);
+    @required this.assignSemantics,
```

Comment:
This looked a little cleaner when an owner was passed in and it would just call the method on that owner...

Reply:
I prefer this style because it ensures isolation so that `_GestureSemantics` will not do any thing fishy (such as modifying properties). I know it's private and we have control but I think it makes code cleaner structurally.